### PR TITLE
Remove deprecated tape execution methods

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
Fix issue #269.

### CODE REVIEW

File: tests/test_adjoint_jacobian.py [hunk 1]
```diff
@@ -320,12 +320,9 @@ def test_gradients(self, op, obs, tol, dev, C):
             qml.expval(obs(wires=0))
             qml.expval(qml.PauliZ(wires=1))
 
-        tape.execute(dev)
+        dev.trainable_params = set(range(1, 1 + op.num_params))
```
Comment (josh146) [inline tests/test_adjoint_jacobian.py:323]: @maliasadi can this line be deleted?

**Review by josh146 — APPROVED** [review]
🎉

**Review by chaeyeunpark — APPROVED** [review]
Nice job @maliasadi

File: tests/test_adjoint_jacobian.py (hunk 1 above)
Comment (maliasadi) [reply tests/test_adjoint_jacobian.py:323]: This is added to the `grad_F` lambda expr.

**Pull request merged**